### DISCUSSION TITLE
机器学习相关问题

### DISCUSSION THREAD

**Discussion by zhaoliang0302** [ShixiangWang/self-study#110] [Q&A] [answered]
翔哥好，

请教一个问题：假设结局变量是二分类变量，而其他用来预测结局变量的predictors也都是二分类，这种时候适合用机器学习方法吗？我自己试了logstic回归、knn、svm、random forest等等方法，发现logstic回归的结果是最好的，不需要调参。而其他机器学习方法再怎么调参也顶多和logistic效果一样。请问predictor和结局变量都是二分类时能弄机器学习吗？谢谢

**Comment by ShixiangWang** [comment, marked as answer]
这个可能有预测能力的上限吧，你可以再试试 xgboost，用 cv 去跑，结果基本是极限了。

**Reply by zhaoliang0302** [reply]
试了xgboost、lightgbm这些，最好的结果和简单的logistic一样，所以我在怀疑这种情况是不是不能用机器学习去做，只能logistic。我自己搜了相关问题没看到这方面解释

**Reply by ShixiangWang** [reply]
可能就是性能极限吧
